apis/nodecore/v1alpha1/common.go

```diff
@@ -43,13 +43,28 @@ type GenericRef struct {
 	// The namespace containing the resource to be referenced. It should be left
 	// empty in case of cluster-wide resources.
 	Namespace string `json:"namespace,omitempty"`
+
+	// The API version of the resource to be referenced.
+	APIVersion string `json:"apiVersion,omitempty"`
+
+	// The kind of the resource to be referenced.
+	Kind string `json:"kind,omitempty"`
+}
+
+func (g *GenericRef) String() string {
+	return fmt.Sprintf("%s:%s - %s/%s", g.APIVersion, g.Kind, g.Namespace, g.Name)
 }
 
 // NodeIdentity is the identity of a FLUIDOS Node.
 type NodeIdentity struct {
-	Domain string `json:"domain"`
-	NodeID string `json:"nodeID"`
-	IP     string `json:"ip"`
+	Domain                string                      `json:"domain"`
+	NodeID                string                      `json:"nodeID"`
+	IP                    string                      `json:"ip"`
+	AdditionalInformation *NodeIdentityAdditionalInfo `json:"additionalInformation,omitempty"`
+}
+
+// NodeIdentityAdditionalInfo contains additional information about the node.
+type NodeIdentityAdditionalInfo struct {
```

Comment:
What do we mean by LiqoID? Is the ClusterID?

Reply:
The LiqoID is the "Liqo Cluster ID", different from the FLUIDOS Node ID or other cluster related ID.

Reply:
> Can we use this to store additional information like Geography etc?

It can be an idea, but we need to keep in mind that we are talking about FLUIDOS Node and not Kubernetes Node, therefore the entire Kubernetes cluster needs to be in the same geographical area.